cas_implementations/src/eav/file.rs

```diff
@@ -170,19 +170,28 @@ impl EntityAttributeValueStorage for EavFileStorage {
         &mut self,
         eav: &EntityAttributeValueIndex,
     ) -> Result<Option<EntityAttributeValueIndex>, HolochainError> {
-        let fetched = self.fetch_eavi(&EaviQuery::new(
-            Some(eav.entity()).into(),
-            Some(eav.attribute()).into(),
-            Some(eav.value()).into(),
-            IndexFilter::LatestByAttribute,
-        ))?;
         let _guard = self.lock.write()?;
-        create_dir_all(self.dir_path.clone())?;
-        let new_eav = increment_key_till_no_collision(eav.clone(), fetched.clone())?;
-        self.write_to_file(ENTITY_DIR.to_string(), &new_eav)
-            .and_then(|_| self.write_to_file(ATTRIBUTE_DIR.to_string(), &new_eav))
-            .and_then(|_| self.write_to_file(VALUE_DIR.to_string(), &new_eav))?;
-        Ok(Some(new_eav.clone()))
+        let wild_card = Path::new("*");
+        //create glob path to query file system parentdir/*/*/*/{address}.txt
+        let text_file = vec![eav.address().to_string(), "txt".to_string()].join(".");
+        let path = self
+            .dir_path
+            .join(wild_card)
+            .join(Path::new("e"))
```

Comment:
Is there any particular reason why you choose to write it this way?
```suggestion
            .join(ENTITY_DIR)
```
should work like a charm

Reply:
Yes that works better, good catch